kernel/src/process.rs

```diff
@@ -655,6 +662,18 @@ impl<C: Chip> ProcessType for Process<'a, C> {
         }
     }
 
+    fn exit(&self, mode: ExitMode, exit_state: State) {
+        match mode {
+            ExitMode::Exit => {
+                self.terminate(State::Exited);
+            }
+            ExitMode::Restart => {
+                static ALWAYS_RESTART: AlwaysRestart = AlwaysRestart::new();
```

Comment:
I don't think that apps should be able to circumvent the kernel settings for restart. The `restart_app` function probably shouldn't take a policy as an argument (the policy should already be attached to the process somehow).

Also, we should open a new PR to move the restart code to its own function, and then the 2.0 change will be a little more compact.